src/tree/values/nodes/TextContainer.js

```diff
@@ -0,0 +1,75 @@
+/**
+ * Represents a text (with optional formatting element(s)) within an HTML-document that can be read by a reader.
+ *
+ * Example:
+ * ```
+ * This text is <strong id="elem-id">very strong</strong>.
+ * ```
+ * should be transformed to:
+ * ```
+ * TextContainer {
+ *     text: "This text is very strong".
+ *     formatting: [
+ *         FormattingElement {
+ *             tag: "strong",
+ *             start: 13,
+ *             end: 24,
+ *             attributes: {
+ *                 id: "elem-id"
+ *             }
+ *             selfClosing: false,
+ *         }
+ *     ]
+ * }
+ * ```
+ */
+class TextContainer {
+	/**
+	 * A piece of text (with optional formatting element(s)) within an HTML-document that can be read by a reader.
+	 *
+	 * @param {string} text                    The plain text, without any formatting.
+	 * @param {FormattingElement[]} formatting The inline HTML-elements (e.g. <strong>, <a>) within the text.
+	 */
+	constructor( text, formatting ) {
+		this.text = text;
+		this.formatting = formatting;
+		this._validate();
+	}
+
+	/**
+	 * Checks if this is a valid TextContainer. If it is not, throws an error.
+	 *
+	 * @private
```

Comment:
Same as with FormattingElement.

JSDoc has an @throws too 😄 
Plus the position could be auto-adjusted, warning instead, possibilities!

Reply:
Changed the code such that it tries to fix possible mistakes and give a warning in the console.